lib/types/array.js

```diff
@@ -341,7 +341,8 @@ module.exports = Any.extend({
                 }
 
                 if (ordereds.length) {
-                    internals.checkRemainingOrdereds(schema, errors, ordereds, value, state, prefs);
+                    internals.fillOrderedErrors(schema, errors, ordereds, value, state, prefs);
+                    internals.fillDefaults(ordereds, value, state, prefs);
```

Comment:
`fillDefaults()` needs to be called after the return statement to make sure all errors are returned before filling defaults.

Reply:
```suggestion
                    }
                    
                    if (errors.length) {
                        return errors;
                    }
                    
                    internals.fillDefault(ordereds, value, state, prefs);
                    return value;
```